daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/BitsCharsetDefinitionRegistry.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.daffodil.processors.charset
+
+import org.apache.daffodil.util.SimpleNamedServiceLoader
+
+/*
+ * Finds all pluggable BitCharsets and makes them available to Daffodil after they have been
+ * setup as described in BitsCharsetDefinition.scala
+ */
+object BitsCharsetDefinitionRegistry {
+
+  private lazy val bitsCharsetDefinitionMap: Map[String, BitsCharsetDefinition] =
+    SimpleNamedServiceLoader.loadClass[BitsCharsetDefinition](classOf[BitsCharsetDefinition])
+
+  /**
+   * Given name, finds the BitsCharsetDefinition or null if not found
+   */
+  def find(name: String): Option[BitsCharsetDefinition] = bitsCharsetDefinitionMap.get(name)
```

Comment:
This doesn't do any case insensitive stuff. I thought we supported lower or upper case charset names? But maybe I'm wrong? Should we, or should all encodings match exactly what is defined by the `name` function returned by the definition?

Reply:
I have confirmed that we do currently support case insensitive encodings. I actually don't see in the code where we handle that, but creating schemas with different cases of the same encoding does work as expected. I'm not sure if we need to do anything special with these new changes or if will just work.

Reply:
Ah, I found how we handle case insensitivity:

https://github.com/apache/daffodil/blob/main/daffodil-lib/src/main/scala/org/apache/daffodil/cookers/Cookers.scala#L82=

The encoding property is always capitalized by our encoding cooker. So in order for a charset to be findable, the name() function must return an all capitalized string. Ours all currently do that I think, but we should probably document that in the `BitsCharsetDefinition` class.

Reply:
I will add that as a comment